Sources/SwiftFusion/Inference/ChordalInitialization.swift

```diff
@@ -0,0 +1,205 @@
+    }
+    
+    for factor in graph.factors(type: PriorFactor3.self) {
+      pose3Graph.store(BetweenFactor3(anchorId, factor.edges.head, factor.prior))
+    }
+    
+    return pose3Graph
+  }
+  
+  /// Construct the orientation graph (unconstrained version of the between in Frobenius norm).
+  public func buildLinearOrientationGraph(
+    g: FactorGraph,
+    v: VariableAssignments,
+    v_linear: VariableAssignments,
+    associations: Dictionary<Int, TypedID<Vector9, Int>>
+  ) -> GaussianFactorGraph {
+    var linearGraph = GaussianFactorGraph(zeroValues: v_linear.tangentVectorZeros)
+
+    for pose3Between in g.factors(type: BetweenFactor3.self) {
+      var Rij = Matrix3.Identity
+
+      Rij = pose3Between.difference.rot.coordinate.R
+      
+      let key1 = associations[pose3Between.edges.head.perTypeID]!
+      let key2 = associations[pose3Between.edges.tail.head.perTypeID]!
+      
+      let M9: Jacobian9x9_2 = Array9([
+        Tuple2(Vector9(-1,0,0,0,0,0,0,0,0), Vector9(Rij.s00,Rij.s01,Rij.s02,0,0,0,0,0,0)),
+        Tuple2(Vector9(0,-1,0,0,0,0,0,0,0), Vector9(Rij.s10,Rij.s11,Rij.s12,0,0,0,0,0,0)),
+        Tuple2(Vector9(0,0,-1,0,0,0,0,0,0), Vector9(Rij.s20,Rij.s21,Rij.s22,0,0,0,0,0,0)),
+        Tuple2(Vector9(0,0,0,-1,0,0,0,0,0), Vector9(0,0,0,Rij.s00,Rij.s01,Rij.s02,0,0,0)),
+        Tuple2(Vector9(0,0,0,0,-1,0,0,0,0), Vector9(0,0,0,Rij.s10,Rij.s11,Rij.s12,0,0,0)),
+        Tuple2(Vector9(0,0,0,0,0,-1,0,0,0), Vector9(0,0,0,Rij.s20,Rij.s21,Rij.s22,0,0,0)),
+        Tuple2(Vector9(0,0,0,0,0,0,-1,0,0), Vector9(0,0,0,0,0,0,Rij.s00,Rij.s01,Rij.s02)),
+        Tuple2(Vector9(0,0,0,0,0,0,0,-1,0), Vector9(0,0,0,0,0,0,Rij.s10,Rij.s11,Rij.s12)),
+        Tuple2(Vector9(0,0,0,0,0,0,0,0,-1), Vector9(0,0,0,0,0,0,Rij.s20,Rij.s21,Rij.s22))
+      ])
+      
+      let b = Vector9(0, 0, 0, 0, 0, 0, 0, 0, 0)
+      
+      linearGraph.store(JacobianFactor9x9_2(jacobian: M9, error: b, edges: Tuple2(key1, key2)))
+    }
+    
+    let I_9x9: Jacobian9x9_1 = Array9([
+      Tuple1(Vector9(1,0,0,0,0,0,0,0,0)),
+      Tuple1(Vector9(0,1,0,0,0,0,0,0,0)),
+      Tuple1(Vector9(0,0,1,0,0,0,0,0,0)),
+      Tuple1(Vector9(0,0,0,1,0,0,0,0,0)),
+      Tuple1(Vector9(0,0,0,0,1,0,0,0,0)),
+      Tuple1(Vector9(0,0,0,0,0,1,0,0,0)),
+      Tuple1(Vector9(0,0,0,0,0,0,1,0,0)),
+      Tuple1(Vector9(0,0,0,0,0,0,0,1,0)),
+      Tuple1(Vector9(0,0,0,0,0,0,0,0,1))
+    ])
+    
+    // prior on the anchor orientation
+    linearGraph.store(JacobianFactor9x9_1(jacobian: I_9x9,
+                                          error: Vector9(1.0, 0.0, 0.0, /*  */ 0.0, 1.0, 0.0, /*  */ 0.0, 0.0, 1.0),
+                                          edges: Tuple1(associations[anchorId.perTypeID]!)))
+    
+    return linearGraph
+  }
+  
+  /// This function finds the closest Rot3 to the unconstrained 3x3 matrix with SVD.
+  /// TODO(fan): replace this with a 3x3 specialized SVD instead of this generic SVD (slow)
+  public func normalizeRelaxedRotations(
```

Comment:
Wherever possible, could you try to mimick the GTSAM organization? You tend to make large verbose loops rather than rely on functions, like "ClosestTo" which localize the functionality in the right place - in this case should be in Rot3

Reply:
done